drivers/SmartThings/philips-hue/src/init.lua

```diff
@@ -55,12 +55,18 @@ local function emit_light_status_events(light_device, light)
     if light.status then
       if light.status == "connected" then
         light_device:online()
+        light_device:set_field(Fields.IS_ONLINE, true)
       elseif light.status == "connectivity_issue" then
         light_device:offline()
+        light_device:set_field(Fields.IS_ONLINE, false)
         return
       end
     end
 
+    if not light_device:get_field(Fields.IS_ONLINE) then
```

Comment:
I would like to understand the consequences of the difference between 

```
if not is_online
```
and
```
if is_online is false
```

Reply:
I was actually debating on making this an explicit boolean check, so I'm glad you asked that question, cuz it's forcing me to think about it harder.

`if not <val>` will also capture the `nil` case, e.g., the value hasn't been set yet, e.g., we haven't had a status update for that bulb in any capacity. I figured it was good to treat that the same as the false case and so treat that bulb as offline. So `if not <val>` will only happen if value is `false` or `nil`; `true` or any other value will evaluate truthy.

`if <val> == false` explicitly looks for that value being assigned false, meaning in the case that we don't know what's going on, we'd treat this as `true`.

If anything, what we may want to do is make an explicit check against the value of `true`, which is what I was going back and forth on.